### DISCUSSION TITLE
Error code: libcasadi_nlpsol_ipopt.so : cannot open shared object file: No such file or directory

### DISCUSSION THREAD

**Discussion by bigbigpark** [casadi/casadi#3086] [Q&A]
Hi,

I'm trying to install casadi library on Jetson nano which is based on Ubuntu 18.04.
I'm using python 2.7 and python 3.6 both.

To install it, I typed below command.

~~~bash
$ pip install pathlib numpy
$ pip install casadi
~~~

And I wrote some short example about `IPOPT`.

~~~python
# test.py
from casadi import *

x = SX.sym("x")
y = SX.sym("y")
z = SX.sym("z")

f = x**2 + 100*z**2
g = z +(1-x)**2 - y
nlp = {'x':vertcat(x,y,z), 'f':f, 'g':g}

solver = nlpsol("solver", "ipopt", nlp)

res = solver(x0   = [2.5, 3.0, 0.75],
                    ubg = 0,
                    lbg = 0)
~~~

I run that code and error occured..

~~~bash
$ python test.py
~~~

~~~bash
CasADi - 2023-04-07 20:55:19 WARNING(".../casadi/core/casadi_os.cpp:166: Assertion "handle!=nullptr" failed:
PluginInterface::load_plugin: Cannot load shared library 'libcasadi_nlpsol_ipopt.so': 
   (
    Searched directories: 1. casadipath from GlobalOptions
                          2. CASADIPATH env var
                          3. PATH env var (Windows)
                          4. LD_LIBRARY_PATH env var (Linux)
                          5. DYLD_LIBRARY_PATH env var (osx)
    A library may be 'not found' even if the file exists:
          * library is not compatible (different compiler/bitness)
          * the dependencies are not found
   )
  Tried '/home/mrl/.local/lib/python2.7/site-packages/casadi' :
    Error code: /home/mrl/.local/lib/python2.7/site-packages/casadi/libcasadi_nlpsol_ipopt.so: cannot open shared object file: No such file or directory
  Tried '' :
    Error code: libcasadi_nlpsol_ipopt.so: cannot open shared object file: No such file or directory
  Tried '.' :
    Error code: ./libcasadi_nlpsol_ipopt.so: cannot open shared object file: No such file or directory") [.../casadi/core/plugin_interface.hpp:142]
Traceback (most recent call last):
  File "test.py", line 43, in <module>
    solver = nlpsol("solver", "ipopt", nlp)
  File "/home/mrl/.local/lib/python2.7/site-packages/casadi/casadi.py", line 28915, in nlpsol
    return _casadi.nlpsol(*args)
RuntimeError: .../casadi/core/plugin_interface.hpp:292: Plugin 'ipopt' is not found.
~~~

Can you help me...?

Thanks.

**Comment by jgillis** [comment]
Hi there,

If I'm not mistaking, pip will have picked up the 3.6.0 source build package, evident from a long installation time?

See https://github.com/casadi/casadi/wiki/FAQ:-how-do-I-configure-a-pip-source-build%3F how to get Ipopt working.

We had experimental aarch64 builds in 3.5.5, but I failed to get them to work yet in 3.6.0..

**Reply by bigbigpark** [reply]
Thanks for fast replying.

As you mentioned when I'm installing `casadi` with `pip install casadi` it had a very long installation time on Jetson Nano.
Alongs to [https://github.com/casadi/casadi/wiki/FAQ:-how-do-I-configure-a-pip-source-build%3F](https://github.com/casadi/casadi/wiki/FAQ:-how-do-I-configure-a-pip-source-build%3F) and you, I do that like

~~~bash
$ pip uninstall casadi
$ export CASADI_SETUP_CMAKE_ARGS="-DWITH_IPOPT=ON\n-DWITH_BUILD_IPOPT=ON"
$ pip install --verbose casadi==3.5.5 # 
~~~

you said experimental aarch64 builds in 3.5.5 and failed the fault version 3.6.0

But Same error occured.

Is there any way I can use `casadi` on Jetson nano?

**Reply by bigbigpark** [reply]
Hi, I found the solution how to install `casadi` on Jetson Nano successfully and to build some optimization frame work `ipopt`.

These are the steps of mine

~~~bash
$ pip uninstall casadi
~~~
Open your bash file, and add following line
~~~bash
$ export CASADI_SETUP_CMAKE_ARGS='-DWITH_IPOPT=ON\n-DWITH_BUILD_IPOPT=ON'
$ export OPENBLAS_CORETYPE=ARMV8
~~~
Install `casadi` 3.5.5 version
~~~bash
$ pip install --verbose casadi==3.5.5
~~~

It worked well for me :)
